operators/pkg/controller/elasticsearch/driver/upgrade_test.go

```diff
@@ -168,9 +178,35 @@ func Test_defaultDriver_doRollingUpgrade(t *testing.T) {
 					green: false,
 				},
 			},
-			want:             newResults().WithResult(defaultRequeue),
+			want:             success().WithResult(defaultRequeue),
 			wantNewPartition: map[string]int32{},
 		},
+		{
+			name: "partially rolled out upgrade",
+			args: args{
+				statefulSets: sset.StatefulSetList{
+					nodespec.TestSset{
+						Name:      "default",
+						Replicas:  2,
+						Data:      true,
+						Partition: 1, // first node has been upgraded
+						Status: appsv1.StatefulSetStatus{
+							CurrentRevision: "a",
+							UpdateRevision:  "b",
+						},
+					}.Build(),
+				},
+				esState: defaultESState,
+			},
+			upgradedPods: map[string]bool{
+				"default-1": true, //
```

Comment:
so much suspense

Reply:
it's like the three dots in a chat window when the other person types ...